packages/EasyDoctrine/docs/doctrine/entity_manager_decorator.md

```diff
@@ -0,0 +1,124 @@
+---eonx_docs---
+title: EntityManager Decorator
+weight: 3001
+is_section: true
+---eonx_docs---
+
+## EntityManagerDecorator
+
+Two main features of EntityManagerDecorator are Transactional and Deferred entity event managing (dispatching or clearing).
+
+### Transactional
+
+`$entityManager->transactional(callable $callback)` provide:
+- flush and commit or rollback if something goes wrong
+- report of any thrown exception with [easy-error-handler][1]
```

Comment:
And we remove `eonx-com/easy-error-handler` from the requirements of this package, right? So if an option is enabled, we try to call the ErrorHandler. If it is not installed/registered, we leave this issue for the project devs. @natepage correct?

Reply:
@roman-eonx usually what I do, I require any optional dependencies as `require-dev` because you still need the dependency during your test.

And to avoid the package not to work if this optional dependency isn't installed but the config is enabled, I actually register the optional functionalities within the application ONLY if:

- config set to enabled
- dependency is installed

Technically something like that (for Symfony):

```php
final class EasyDoctrineExtension extends Extension
{
    public function load(array $configs, ContainerBuilder $container): void
    {
          // Resolve your config and loader....

          if (($config['easy_error_handler_enabled'] ?? true) && \interface_exists(ErrorHandlerInterface::class)) {
              $loader->load('easy-error-handler-listener.php');
          }
    }
}
```